write_buffer/src/core.rs

```diff
@@ -168,15 +178,19 @@ pub trait WriteBufferWriting: Sync + Send + Debug + 'static {
 
 /// Handles a stream of a specific sequencer.
 ///
-/// This can be used to consume data via a stream or to seek the stream to a given offset.
+/// This can be used to consume data via a stream or to seek the stream to a given sequence number.
 #[async_trait]
 pub trait WriteBufferStreamHandler: Sync + Send + Debug + 'static {
     /// Stream that produces DML operations.
     ///
     /// Note that due to the mutable borrow, it is not possible to have multiple streams from the same
     /// [`WriteBufferStreamHandler`] instance at the same time. If all streams are dropped and requested again, the last
-    /// offsets of the old streams will be the start offsets for the new streams. If you want to prevent that either
-    /// create a new [`WriteBufferStreamHandler`] or use [`seek`](Self::seek).
+    /// sequence number of the old streams will be the start sequence number for the new streams. If you want to
+    /// prevent that either create a new [`WriteBufferStreamHandler`] or use [`seek`](Self::seek).
+    ///
+    /// If the sequence number that the stream wants to read is unknown (either because it is in the future or because
+    /// some retention policy removed it already), the stream will return an error w/
```

Comment:
```suggestion
    /// some retention policy removed it already), the stream will return an error with
```

I would avoid shorthand in public API docs.